### PR TITLE
update setjmp

### PR DESCRIPTION
**Pull request by lukileczo**
## Description

- Add arch-dependent `jmp_buf` definition
- Save FP registers on `riscv64`
- `arm`: align `.text` to 4, change `ldr/str` to `ldm/stm` to save 2 bytes each
## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## How Has This Been Tested?

- [x] Already covered by automatic testing.
- [ ] New test added: (add PR link here).
- [ ] Tested by hand on: (list targets here).

## Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [x] All new and existing linter checks and tests passed.
- [x] My changes generate no new compilation warnings for any of the targets.

## Special treatment

- [ ] This PR needs additional PRs to work (list the PRs, preferably in merge-order).
- [ ] I will merge this PR by myself when appropriate.

### CODE REVIEW

File: arch/armv7a/jmp.S [hunk 1]
```diff
@@ -14,29 +14,28 @@
...
-  * 80      fpscr register
-  */
+/* setjmp env buffer description
+ * Offset  Description
+ * 0       Signal mask flag - if != 0 then next value stores a signal mask
+ * 4       Signal mask
+ * 8       SP register
+ * 12-43   r4-r11 registers
+ * 44      LR register
+ * 48-103  d8-d15 registers
```
Comment (agkaminski) [inline arch/armv7a/jmp.S:24]: Hmm, I think it was correct before:
- 48 d8
- 52 d9
- 56 d10
- 60 d11
- 64 d12
- 68 d13
- 72 d14
- 76 d15
- 80 fpscr

File: arch/armv7m/jmp.S [hunk 2]
```diff
@@ -14,29 +14,28 @@
...
-  * 80      fpscr register
-  */
+/* setjmp env buffer description
+ * Offset  Description
+ * 0       Signal mask flag - if != 0 then next value stores a signal mask
+ * 4       Signal mask
+ * 8       SP register
+ * 12-43   r4-r11 registers
+ * 44      LR register
+ * 48-103   d8-d15 registers
```
Comment (agkaminski) [inline arch/armv7m/jmp.S:24]: As with armv7a, imho the previous one was correct. One space needs to be removed

Comment (lukileczo) [reply arch/armv7m/jmp.S:24]: hmm, as per the [armv7-m manual](https://developer.arm.com/documentation/ddi0403/d/Application-Level-Architecture/Application-Level-Programmers--Model/The-optional-Floating-point-extension/The-FP-extension-registers?lang=en):
> Software can access the FP extension register bank as:
> - thirty-two 32-bit single-precision registers, S0-S31
> - sixteen 64-bit doubleword registers, D0-D15.

So these are 8-byte each

Comment (agkaminski) [reply arch/armv7m/jmp.S:24]: Right, sorry

**Review by agkaminski — APPROVED** [review]

**Pull request merged**